test/composing_test.exs

```diff
@@ -5,75 +5,93 @@ defmodule Membrane.VideoCompositor.ComposingTest do
-        second_raw_video_path: "./test/fixtures/short_videos/input_3s_720p_1fps.raw"
-      }
-
-      output_path = Path.join(ctx.tmp_dir, "output_3s_1280x1440_1fps.raw")
-      composed_video_path = "./test/fixtures/short_videos/composed_video_3s_1280x1440_1fps.raw"
-
-      video_caps = %RawVideo{
-        width: 1280,
-        height: 720,
-        framerate: {1, 1},
-        pixel_format: :I420,
-        aligned: nil
-      }
-
-      implementation = :nx
-
-      test_raw_pipeline_and_composing(
-        input_paths,
-        output_path,
-        composed_video_path,
-        video_caps,
-        implementation
-      )
-    end
-
-    @tag long: true
-    test "10s 720p 1fps raw video", ctx do
-      input_paths = %{
-        first_raw_video_path: "./test/fixtures/long_videos/input_10s_720p_1fps.raw",
-        second_raw_video_path: "./test/fixtures/long_videos/input_10s_720p_1fps.raw"
-      }
-
-      output_path = Path.join(ctx.tmp_dir, "output_10s_1280x1440_1fps.raw")
-      composed_video_path = "./test/fixtures/long_videos/composed_video_10s_1280x1440_1fps.raw"
-
-      video_caps = %RawVideo{
-        width: 1280,
-        height: 720,
-        framerate: {1, 1},
-        pixel_format: :I420,
-        aligned: nil
-      }
-
-      implementation = :nx
-
-      test_raw_pipeline_and_composing(
-        input_paths,
-        output_path,
-        composed_video_path,
-        video_caps,
-        implementation
-      )
+  alias Membrane.VideoCompositor.Test.Utility, as: TestingUtility
+
+  @filter_description "split[b], pad=iw:ih*2[src], [src][b]overlay=0:h"
+  @implementations [:nx, :ffmpeg]
+
+  Enum.map(@implementations, fn implementation ->
+    describe "Checks #{implementation} composition and raw video pipeline on" do
+      @describetag :tmp_dir
+
+      test "3s 720p 1fps raw video", %{tmp_dir: tmp_dir} do
+        video_caps = %RawVideo{
+          width: 1280,
+          height: 720,
+          framerate: {1, 1},
+          pixel_format: :I420,
+          aligned: nil
+        }
+
+        duration = 3
+
+        test_raw_pipeline_and_composing(
+          video_caps,
+          duration,
+          unquote(implementation),
+          tmp_dir,
+          "short_videos"
+        )
+      end
+
+      @tag long: true
+      test "10s 720p 1fps raw video", %{tmp_dir: tmp_dir} do
+        video_caps = %RawVideo{
+          width: 1280,
+          height: 720,
+          framerate: {1, 1},
+          pixel_format: :I420,
+          aligned: nil
+        }
+
+        duration = 10
+
+        test_raw_pipeline_and_composing(
+          video_caps,
+          duration,
+          unquote(implementation),
+          tmp_dir,
+          "long_videos"
+        )
+      end
     end
-  end
-
+  end)
+
+  @spec test_raw_pipeline_and_composing(
+          Membrane.RawVideo.t(),
+          non_neg_integer(),
+          atom,
+          binary(),
+          binary()
+        ) :: nil
```

Comment:
Specs are unecessary for functions in tests (especially in private ones), but we can leave it here